mantis-network/src/main/java/io/reactivex/mantis/network/push/PushServer.java

```diff
@@ -235,128 +245,174 @@ protected Observable<Void> manageConnectionWithCompression(final DefaultChannelW
         final BasicTag slotIdTag = new BasicTag("slotId", slotId);
 
         SerializedSubject<List<byte[]>, List<byte[]>> subject
-                = new SerializedSubject<>(PublishSubject.<List<byte[]>>create());
+            = new SerializedSubject<>(PublishSubject.<List<byte[]>>create());
         Observable<List<byte[]>> observable = subject.lift(new DropOperator<>("batch_writes", slotIdTag));
 
         if (applySampling) {
             observable =
-                    observable
-                            .sample(samplingRateMSec, TimeUnit.MILLISECONDS)
-                            .map((List<byte[]> list) -> {
-                                        // get most recent item from sample
-                                        List<byte[]> singleItem = new LinkedList<>();
-                                        if (!list.isEmpty()) {
-                                            singleItem.add(list.get(list.size() - 1));
-                                        }
-                                        return singleItem;
-                                    }
-                            );
+                observable
+                    .sample(samplingRateMSec, TimeUnit.MILLISECONDS)
+                    .map((List<byte[]> list) -> {
+                            // get most recent item from sample
+                            List<byte[]> singleItem = new LinkedList<>();
+                            if (!list.isEmpty()) {
+                                singleItem.add(list.get(list.size() - 1));
+                            }
+                            return singleItem;
+                        }
+                    );
+        }
+
+        final BasicTag clientIdTag = new BasicTag(CLIENT_ID_TAG_NAME, Optional.ofNullable(groupId).orElse("none"));
+        Metrics writableMetrics = new Metrics.Builder()
+            .id("PushServer", clientIdTag)
+            .addCounter("channelWritable")
+            .addCounter("channelNotWritable")
+            .addCounter("channelNotWritableTimeout")
+            .build();
+        metricsRegistry.registerAndGet(writableMetrics);
+        Counter channelWritableCounter = writableMetrics.getCounter("channelWritable");
+        Counter channelNotWritableCounter = writableMetrics.getCounter("channelNotWritable");
+        Counter channelNotWritableTimeoutCounter = writableMetrics.getCounter("channelNotWritableTimeout");
+
+        final Future<?> writableCheck;
```

Comment:
@sundargates  These lines are the only ones that are relevant. Everything else is not important.